### PR TITLE
feat: add `.parent` field

### PR DESCRIPTION
**Pull request by saschanaz**
WIP because I want everything to be less complex than this.

This patch includes:
- [x] A relevant test
- [x] A relevant documentation update

### CODE REVIEW

**Comment by saschanaz** [comment]
I think the patch is now simple enough, but somewhat magical... What do you think, @marcoscaceres ?

File: README.md [hunk 1]
```diff
@@ -550,6 +535,7 @@ The fields are as follows:
 * `name`: The name of the operation if exists.
 * `arguments`: An array of [arguments](#arguments) for the operation.
 * `extAttrs`: An array of [extended attributes](#extended-attributes).
+* `parent`: The container of this type.
```
Comment (marcoscaceres) [inline README.md]: Maybe: 

```suggestion
* `parent`: The container of this type as an Object.
```

Or anything else that might aid in the usage. I guess they are of whatever parent type. I guess this comment applies to the rest of the documentation.

File: lib/productions/helpers.js [hunk 2]
```diff
@@ -242,3 +242,43 @@ export function findLastIndex(array, predicate) {
...
+
+/**
+ * Returns a proxy that auto-assign `parent` field.
+ * @template T
+ * @param {T} data
+ * @param {*} parent this is `data` by default
+ * @return {T}
+ */
+export function autoParenter(data, parent) {
+  if (!parent) {
```
Comment (marcoscaceres) [inline lib/productions/helpers.js]: Might be nice to document what's happening here (and below).... this is really nice,  but you are right that it's a bit magical. I'd be a bit scared to come in here in the future if anything goes wrong.

File: lib/productions/helpers.js [hunk 3]
```diff
@@ -242,3 +242,43 @@ export function findLastIndex(array, predicate) {
...
+    parent = data;
+  }
+  if (!data || typeof data !== "object") {
```
Comment (marcoscaceres) [inline lib/productions/helpers.js]: Similarly here... explaining why this is not "exceptional" would be good (I guess this is parent root or something).

File: lib/productions/helpers.js [hunk 4]
```diff
@@ -242,3 +242,43 @@ export function findLastIndex(array, predicate) {
...
+    return data;
+  }
+  return new Proxy(data, {
+    get(target, p) {
+      const value = target[p];
+      if (Array.isArray(value)) {
+        return autoParenter(value, target);
```
Comment (marcoscaceres) [inline lib/productions/helpers.js]: Explaining this one a little bit would be good too... recursive array fun :)

**Pull request merged**